.github/workflows/test.yml

```
@@ -735,3 +720,97 @@ jobs:
           files: ./packages/generator-helper/src/__tests__/coverage/clover.xml
           flags: generator-helper,${{ matrix.os }},library,binary
           name: generator-helper-${{ matrix.os }}
+
+      - name: Upload test results to BuildPulse for flaky test detection
+        if: ${{ !cancelled() }} # Run this step even when the tests fail. Skip if the workflow is cancelled.
```

👋 @janpio: In our email chat, I mentioned that we should only run this step on branches that are in this repository (as opposed to running it in forks, since forks don't have access to secrets). To set this up, we can use use this check (similar to the [approach used in Homebrew/brew](https://github.com/Homebrew/brew/blob/01506f120abd0d258afeb942b56deb3dd82470a9/.github/workflows/tests.yml#L264)):

```suggestion
        # Only run this step for branches where where we have access to secrets.
        # Run this step even when the tests fail. Skip if the workflow is cancelled.
        if: startsWith(github.repository, 'prisma/') && !cancelled()
```